source/common/upstream/upstream_impl.cc

```diff
@@ -495,7 +496,12 @@ void ClusterImplBase::initialize(std::function<void()> callback) {
 }
 
 void ClusterImplBase::onPreInitComplete() {
-  sds_init_manager_.initialize([this]() { onSdsInitDone(); });
+  if (sds_init_manager_) {
+    sds_init_manager_->initialize([this]() { onSdsInitDone(); });
+    sds_init_manager_.reset();
```

Comment:
Add comment on
onPreInitComplete() may be called multiple times,  but sds_init_manager only can be called once.